File: ExpenseRecorder/Models/PaymentAccount.cs
using ExpenseRecorder.Models ;
using ExpenseRecorder.Models.Interfaces ;

public class PaymentAccount : IUserEntity< PaymentAccount >
{
	public int    Id     { get ; set ; }
	public string UserId { get ; set ; } = String.Empty ;


	public string  Name     { get ; set ; } = String.Empty;
	// TODO: may be currency should be a separate entity stored in the database
	public string  Currency { get ; set ; } = String.Empty ;
	public decimal Balance  { get ; set ; }


	public ICollection< Transaction > Transactions { get ; set ; } = null! ;

	public void CopyFrom(PaymentAccount entity)
	{
		Name     = entity.Name ;
		Currency = entity.Currency ;
		Balance  = entity.Balance ;
	}
}

File: ExpenseRecorder/ClientApp/src/app/modules/payment-account/models/payment-account.ts
export interface PaymentAccountItem {
	id : number;
	name : string;
	balance : number;
	currency : string;
}

export interface PaymentAccountViewItem {
	id : number;
	name : string;
	balance : number;
	currency : string;
	transactions : string[];
}

export interface PaymentAccountCreateUpdateItem {
	name : string;
	balance : number;
	currency : string;
}

